### ISSUE TITLE
Use `prepublish` hook instead of `preinstall` to generate the bin

### ISSUE THREAD

**Issue by sorccu** [jsx/JSX#97]
I just happily installed JSX 0.9.2 and saw that it was using a preinstall hook. This is actually not a very good thing. Here's a snippet from `npm help scripts`:

```
NOTE: INSTALL SCRIPTS ARE AN ANTIPATTERN
       tl;dr Don't use install.  Use a .gyp file for com-
       pilation, and prepublish for anything else.

       You should almost never have to explicitly  set  a
       preinstall  or  install  script.  If you are doing
       this, please consider if there is another  option.

       The  only  valid  use  of  install  or  preinstall
       scripts is for compilation which must be  done  on
       the  target  architecture.   In  early versions of
       node, this  was  often  done  using  the  node-waf
       scripts,  or a standalone Makefile, and early ver-
       sions of npm required that it be explicitly set in
       package.json.   This  was not portable, and harder
       to do properly.

       In the current version of node, the  standard  way
       to  do  this  is using a .gyp file.  If you have a
       file with a .gyp extension in  the  root  of  your
       package,   then   npm  will  run  the  appropriate
       node-gyp commands automatically at  install  time.
       This  is  the only officially supported method for
       compiling binary addons, and does not require that
       you add anything to your package.json file.

       If you have to do other things before your package
       is used, in a way that is  not  dependent  on  the
       operating  system  or  architecture  of the target
       system, then  use  a  prepublish  script  instead.
       This includes tasks such as:

       o   Compile    CoffeeScript   source   code   into
           JavaScript.

       o   Create minified versions of JavaScript  source
           code.

       o   Fetching  remote  resources  that your package
           will use.



       The advantage of doing these things at  prepublish
       time instead of preinstall or install time is that
       they can be done once, in a single place, and thus
       greatly  reduce complexity and variability.  Addi-
       tionally, this means that:

       o   You can depend on coffee-script as a devDepen-
           dency,  and thus your users don't need to have
           it installed.

       o   You don't need to  include  the  minifiers  in
           your  package,  reducing  the  size  for  your
           users.

       o   You don't need to rely on  your  users  having
           curl or wget or other system tools on the tar-
           get machines.
```

Because JSX's bin creation does not require anything architecture-specific, it would be MUCH preferred to use a prepublish hook for the `make compiler` phase. But I assume you want to do this so that people can just clone the repository and do `npm install` locally if they want. In this case it would be much preferred to just include the generated bin file in the Git repo.

Also, it would make JSX installable on Windows.

So basically the options are:
1. Use prepublish instead of preinstall. Pros: very small and easy change. Cons: people cloning the repo would have to run `make compiler` manually.
2. Generate and add the bin to Git. Pros: usable right out of the box everywhere. Cons: you will have to remember to keep the bin up to date.
